crates/sui-adapter/src/execution_mode.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_vm_runtime::session::SerializedReturnValues;
+use sui_types::error::ExecutionError;
+
+pub type TransactionIndex = usize;
+
+pub trait ExecutionMode {
+    type ExecutionResult;
+    type ExecutionResults;
+
+    fn allow_arbitrary_function_calls() -> bool;
```

Comment:
A comment w/ the contents from the PR summary explaining what this enables in more detail would be useful.